api/services/namespace.go

```diff
@@ -180,28 +180,18 @@ func (s *service) fillMembersData(ctx context.Context, members []models.Member)
 	return members, nil
 }
 
-// EditNamespace edits the namespace name.
-//
-// It receives a context, used to "control" the request flow,  tenant ID from models.Namespace and the new name to
-// namespace. Name is set to lowercase.
-//
-// EditNamespace returns a models.Namespace and an error. When error is not nil, the models.Namespace is nil.
-func (s *service) EditNamespace(ctx context.Context, tenantID, name string) (*models.Namespace, error) {
-	namespace, err := s.store.NamespaceGet(ctx, tenantID)
-	if err != nil {
-		return nil, NewErrNamespaceNotFound(tenantID, err)
```

Comment:
I believe keeping the namespace get to guarantee the namespaces existence facilitates the reading process

Reply:
Due to the fact that this method returns the updated namespace, I believe it is unnecessary to search for the namespace twice, leading to the following (excessive in my opinion) flow:

1. Retrieve the namespace solely to check for its existence.
2. Update the namespace.
3. Retrieve the updated namespace.

There may also be a third find operation if searching for a duplicated name is necessary.

I also believe it's not overly complicated to understand that updating a namespace assumes it already exists, potentially returning an error if it does not. Additionally, I think that we can leverage the power of MongoDB, which can handle this with only one operation.